plugin-gradle/src/main/java/com/diffplug/gradle/spotless/SpotlessTask.java

```diff
@@ -168,11 +175,27 @@ public void performAction(IncrementalTaskInputs inputs) throws Exception {
 				.steps(steps)
```

Comment:
We do `try(Formatter formatter = ...` at the top.  With this commit, it is going to turn out all the time that `outOfDate.isEmpty()` and we don't use the formatter for anything.  I would wait to see if `outOfDate.isEmpty()` and just return early, possibly saving the creation of the formatter entirely.  Before this commit, `outOfDate` would never be empty, which is why we didn't have this optimization before.

Reply:
updated